modules/infra/submodules/storage/iam.tf

```diff
@@ -56,6 +56,27 @@ data "aws_iam_policy_document" "ecr" {
       "ecr:PutImage"
     ]
   }
+
+  override_policy_documents = local.supports_pull_through_cache ? [data.aws_iam_policy_document.ecr_pull_through_cache[0].json] : []
+}
+
+data "aws_iam_policy_document" "ecr_pull_through_cache" {
+  count = local.supports_pull_through_cache ? 1 : 0
+
+  statement {
+    effect = "Allow"
+
+    resources = [
+      "arn:${data.aws_partition.current.partition}:ecr:${var.region}:${data.aws_caller_identity.this.account_id}:repository/${aws_ecr_pull_through_cache_rule.quay[0].ecr_repository_prefix}/*"
```

Comment:
we have this starting in TF 1.8 
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/functions/arn_build

Reply:
I'd prefer to wait a bit to not bundle this with the TF upgrade since there's a handful of other places we're doing this